ivy_tests/test_ivy/test_functional/test_core/test_elementwise.py

```diff
@@ -2447,3 +2447,37 @@ def test_trunc_divide(
         x1=x[0],
         x2=x[1],
     )
+
+
+# isreal
+@handle_cmd_line_args
+@given(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("real_and_complex")
+    ),
+    num_positional_args=helpers.num_positional_args(fn_name="isreal"),
+)
+def test_isreal(
+    *,
+    dtype_and_x,
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
+    container,
+    instance_method,
+    fw,
+):
+    input_dtype, x = dtype_and_x
+    helpers.test_function(
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=with_out,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        container_flags=container,
+        instance_method=instance_method,
+        fw=fw,
+        fn_name="isreal",
+        x=np.asarray(x, dtype=input_dtype),
```

Comment:
According to the update made recently, `x` received from the helper function is already a numpy array, so we don't need to use `np.asarray` here, please pass `x` directly as the keyword argument